onnx/onnx.in.proto

```diff
@@ -497,6 +497,50 @@ message TensorShapeProto {
   repeated Dimension dim = 1;
 }
 
+// Sequences
+//
+// Defines a dense, ordered, collection of elements that are of homogeneous types.
+message SequenceProto {
+  // Sequences are made of TypeProto elements, which can either be tensors,
+  // maps, or sequences themselves.
+  message SequenceElement {
+    oneof value {
+      optional TensorProto tensor_value = 1;
+      optional SequenceProto sequence_value = 2;
+      optional MapProto map_value = 3;
+    };
+  };
+  repeated SequenceElement values = 1;
+
+  // shape of the underlying sequence
+  repeated int64 dims = 2;
+
+  // The data type of the sequence.
+  optional TypeProto elem_type = 3;
```

Comment:
is this needed? will it be conflict with the type in "TensorProto"?

Reply:
Yes, I was wondering the same. Why do we need the "dims" field also? (I think the edge-case where we have an empty-sequence may need it, depending on the use-case. But ideally we should not need this.)

Reply:
Yes, I added dims originally for the empty-sequence case and for ease of use, but it's perhaps overengineered. I'm removing both elem_type and dims.